system/doc/reference_manual/expressions.xml

```diff
@@ -836,6 +836,20 @@ number &lt; atom &lt; reference &lt; fun &lt; port &lt; pid &lt; tuple &lt; map
       depending on the size of the float because otherwise comparison of large
       floats and integers would lose their transitivity.</p>
 
+    <p>The term equivalence operators, <c>=:=</c> and <c>=/=</c>, return
+      whether two terms are indistinguishable. While the other operators
+      consider the same <em>numbers</em> equal regardless of whether they are
+      floats or integers (<c>1 == 1.0</c> is true), the term equivalence
+      operators return whether there exists any function that can tell their
+      arguments apart.</p>
+    <p>For example, while the terms <c>0.0</c> and <c>-0.0</c> represent the
+      exact same <em>number</em>, they yield different results when converted
+      to string form through <c>float_to_list/1</c>, so <c>=:=</c> and
+      <c>=/=</c> consider them different. This is useful when reasoning about
+      terms as opaque values, for example in associative containers or memoized
+      functions where using the equal-to (<c>==</c>) operator can result in
+      subtly incorrect results.</p>
```

Comment:
To me this explanation explains why `==` and `=:=` behaves differently on `0.0` and `0`, but not why `=:=` regards `0.0` as different from `-0.0`.

Reply:
Where should we explain that `0.0` and `+0.0` are exactly equal in runtime but have slightly different meaning in compile time?

Reply:
>  To me this explanation explains why == and =:= behaves differently on 0.0 and 0, but not why =:= regards 0.0 as different from -0.0.

I've pushed up a commit that I hope will be a bit clearer.

> Where should we explain that 0.0 and +0.0 are exactly equal in runtime but have slightly different meaning in compile time?

I'm not sure that's necessary: they are the same even in compile-time, the compiler just happens to give the user a helpful warning when using the former in a pattern or term equality test.

Reply:
But we need to state that they are the same

Reply:
Sure, I've pushed up another commit to that effect. :)